libraries/chain/db_maint.cpp

```diff
@@ -471,6 +471,18 @@ void database::update_active_sons()
    } else {
       ilog( "Active SONs set CHANGED" );
       // Store new SON info, initiate wallet recreation and transfer of funds
+      for( const son_object& son : cur_active_sons )
```

Comment:
There is a requirement in FR document wherein if a SON is in maintenance mode for more than 12 hours then it should be deregistered by witnesses. If we move all active SONs to inactive we might lose track of the maintenance mode.

We might also have to change the heartbeat evaluator code to check for the state,
i.e. if a SON is in maintenance mode and it sends heartbeat we shouldn't move it straight back to active. We should check if that SON is on the list of active sons in GPO. (gpo.active_sons) after chain maintenance.

Reply:
Good catch.) I'll add the separate commit to fix it.